host/volume/zfs/persistence_test.go

```diff
@@ -0,0 +1,129 @@
+package zfs_test
+
+import (
+	"fmt"
+	"math"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	. "github.com/flynn/flynn/Godeps/_workspace/src/github.com/flynn/go-check"
+	gzfs "github.com/flynn/flynn/Godeps/_workspace/src/github.com/mistifyio/go-zfs"
+	"github.com/flynn/flynn/host/volume"
+	"github.com/flynn/flynn/host/volume/manager"
+	"github.com/flynn/flynn/host/volume/zfs"
+	"github.com/flynn/flynn/pkg/random"
+	"github.com/flynn/flynn/pkg/testutils"
+)
+
+// note: many of these tests are not zfs specific; refactoring this when we have more concrete backends will be wise.
+
+type PersistenceTests struct{}
+
+var _ = Suite(&PersistenceTests{})
+
+func (PersistenceTests) SetUpSuite(c *C) {
+	// Skip all tests in this suite if not running as root.
+	// Many zfs operations require root priviledges.
+	testutils.SkipIfNotRoot(c)
+}
+
+func (s *PersistenceTests) TestApplePie(c *C) {
```

Comment:
Creating the universe from scratch :)

Kidding, placeholder needs rename badly.